text/0517-io-os-reform.md

```diff
@@ -600,11 +609,11 @@ The `Writer` trait is cut down to even smaller size:
 
 ```rust
 trait Write {
-    fn write(&mut self, buf: &[u8]) -> Result<uint, Error>;
+    fn write(&mut self, buf: &[u8]) -> Result<usize, Error>;
     fn flush(&mut self) -> Result<(), Error>;
 
-    fn write_all(&mut self, buf: &[u8]) -> Result<(), Error> { .. }
-    fn write_fmt(&mut self, fmt: &fmt::Arguments) -> Result<(), Error> { .. }
+    fn write_all(&mut self, buf: &[u8]) -> Result<usize, Error> { .. }
```

Comment:
This is a crucial part of the design of `write_all` that it *fails* if the entire buffer could not be written. Otherwise there are no real semantic wins over the `write` function and this ends up just largely being a duplicate.